Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialTypeSourceData.cpp

```diff
@@ -32,6 +32,40 @@ namespace AZ
         namespace
         {
             [[maybe_unused]] static constexpr char const MaterialTypeSourceDataDebugName[] = "MaterialTypeSourceData";
+
+            void SortMaterialTypeSourceDataPropertyGroups(
+                AZStd::vector<AZStd::unique_ptr<MaterialTypeSourceData::PropertyGroup>>& propertyGroups)
+            {
+                // Sort child groups alphabetically
+                AZStd::sort(
+                    propertyGroups.begin(),
+                    propertyGroups.end(),
+                    [](auto& group1, auto& group2)
+                    {
+                        return group1->GetName() < group2->GetName();
+                    });
+
+                // Sort general groups to the back of the list
+                AZStd::stable_sort(
+                    propertyGroups.begin(),
+                    propertyGroups.end(),
+                    [](auto& group1, auto& group2)
+                    {
+                        static constexpr AZStd::string_view generalGroupName = "general";
+                        const auto& groupName1 = group1->GetName();
+                        const auto& groupName2 = group2->GetName();
+                        return AZ::StringFunc::Equal(groupName1, generalGroupName) < AZ::StringFunc::Equal(groupName2, generalGroupName);
+                    });
```

Comment:
It seems like these two sorts can be combined where the any general group name check occurs first.